src/cljam/io/sam/util/cigar.clj

```diff
@@ -98,6 +98,12 @@
           (recur (+ ref-length (case op 0 n 2 n 3 n 7 n 8 n 0))))
         [(.toString sb) ref-length]))))
 
+(defn placeholder?
```

Comment:
This function is used to determine if the CIGAR block to be decoded has been replaced by a placeholder. 
I have adopted the implementation of https://github.com/chrovis/cljam/pull/263#discussion_r994202988. This is concise and performs well.

This implementation is not completely according to the specification of samtools. 
The differences are as follows :
In the samtools specification, the first operation of cigar is `kS` (`k` length, Soft-clipping), and it determines if `k` is equal to `l_seq`. 
This implementation does not consider the length of the operation; it only determines if the CIGAR format matches `kSmN`.

I and @alumi  assume that this implementation is fine for practical use. 
This is because :
- Since samtools started supporting long cigar encoding, the specification of encoding with `kSmN` format and creating optional CG tags has been consistent.
  - Therefore, in the long CIGAR placeholder, it is assumed that  anything other than `kSmN` need not be considered.
- Alignments whose cigar matches the form `kSmN` and, `k` and `l_seq` are not equal are considered invalid. 
  - Therefore, as long as we are dealing with normal data, it would not make much sense to examine the operation length.

Reply:
Thanks for elaborating on how the current implementation works and how it differs from the samtools implementation 👍 
 It would help for future reference if you also clarify why you chose this implementation in spite of the differences. I guess it's because it’s performant and easy to implement.

Reply:
Thank you for your advice! 🙏 
Just as you said, I implemented it this way because this is concise and performs well.